File: 2023/UICPC Regional/Round #15/solutions/hothike/per.rb
# I have no idea what I'm doing.

nums = STDIN.read.split().map(&:to_i)
n = nums[0]
d = 0
t = 50
for i in 1..n-2
  if [nums[i], nums[i+2]].max < t
    d = i
    t = [nums[i], nums[i+2]].max
  end
end
puts d.to_s + " " + t.to_s

File: 2023/UICPC Regional/Round #15/solutions/hothike/per.go
// I have no idea what I'm doing.

package main

import (
    "fmt"
)

func main() {
    var n, d, t int
    var T [100]int

    fmt.Scanf("%d", &n)

    for i := 0; i < n; i++ {
        fmt.Scanf("%d", &T[i])
    }
    t = 50
    for i := 0; i < n-2; i++ {
        var v int
        v = T[i]
        if T[i+2] > v {
           v = T[i+2]
        }
        if v < t {
           d = i+1
           t = v
        }
    }
    fmt.Printf("%d %d\n", d, t)
}
